app/javascript/src/Types/SwaggerTypes.ts

```diff
@@ -17,3 +18,51 @@ export interface ApiDocsServices {
   host: string;
   apis: ApiDocsService[];
 }
+
+export interface RequestData extends Request {
+  url: string;
+  method: SupportedHTTPMethods;
+  body: string;
+  credentials: string;
+  headers: Record<string, string>;
+  requestInterceptor?: ((request: Request) => Promise<Request> | Request) | undefined;
+  responseInterceptor?: ((response: Response) => Promise<Response> | Response) | undefined;
+}
+
+export interface ExecuteData {
+  contextUrl: string;
+  fetch: (arg: unknown) => unknown;
+  method: SupportedHTTPMethods;
+  operation: unknown;
+  operationId: string;
+  parameters: unknown;
+  pathName: string;
+  requestBody: unknown;
+  requestContentType: string;
+  requestInterceptor?: ((request: Request) => Promise<Request> | Request) | undefined;
+  responseContentType: string;
+  responseInterceptor?: ((response: Response) => Promise<Response> | Response) | undefined;
+  scheme: string;
+  securities: unknown;
+  server: string;
+  serverVariables: unknown;
+  spec: unknown;
+}
```

Comment:
The rest are utility types but I think this one strictly belongs to `swagger-client` and it should be defined in swagger.d.ts.